src/components/roomResults/RoomCardModal/RoomCardModal.tsx

```diff
@@ -0,0 +1,184 @@
+          }}
+        >
+          <Box>
+            <Box sx={{ display: "flex", gap: "2.18rem" }}>
+              <Box sx={{ display: "flex", gap: "0.3rem" }}>
+                <PersonOutlineIcon sx={{ color: "#5d5d5d" }} />
+                <Typography fontSize={"0.875rem"} color={"#5D5D5D"} fontWeight={400}>
+                  1-{room.maxCapacity}
+                </Typography>
+              </Box>
+              <Box sx={{ display: "flex", gap: "0.4rem" }}>
+                <BedIcon sx={{ color: "#5d5d5d" }} />
+                <Typography fontSize={"0.875rem"} color={"#5D5D5D"} fontWeight={400}>
+                  {room.bedType}
+                </Typography>
+              </Box>
+              <Box sx={{ display: "flex", gap: "0.3rem" }}>
+                <Typography fontSize={"0.875rem"} color={"#5D5D5D"} fontWeight={400}>
+                  {`${room.areaInSquareFeet}ft`}
+                </Typography>
+              </Box>
+            </Box>
+            <Typography mt={"1.375rem"}>{roomType[`${room.roomTypeName}`].description}</Typography>
+
+            {/* ------------------------------------------------------- Promotions ---------------------------------------------------- */}
+            <Box mt={"1.625rem"} className="standard-rate">
+              <Typography fontSize={"1.25rem"} fontWeight={"700"}>
+                Standard Rate
+              </Typography>
+              {/* --------------------------------------- Promotion Card ----------------------------------------------------------- */}
+              <PromotionCard />
+            </Box>
+            {/* ---------------------------------------------------------------- Deals And Packages ------------------------------------- */}
+            <Box mt={"2.5rem"} className="deals-packages">
+              <Typography fontSize={"1.25rem"} fontWeight={"700"}>
+                Deals & Packages
+              </Typography>
+              <PromotionCard />
+              <PromotionCard />
+            </Box>
+            {/* --------------------------------------------------------------- Promo Code ------------------------------------------------ */}
+            <Box mt={"2.5rem"}>
+              <Typography>Enter a promo code</Typography>
+              <Box
+                mt={"5px"}
+                sx={{
+                  display: "grid",
+                  gridTemplateColumns: "0.7fr 0.3fr",
+                  gap: "0.813rem",
+                  maxWidth: "20rem",
+                }}
+              >
+                <TextField id="outlined-basic" variant="outlined" />
+                <Button
+                  type="submit"
+                  sx={{
+                    backgroundColor: "#26266D",
+                    "&:hover": { backgroundColor: "#26266D" },
+                    padding: "0.75rem 1.25rem",
+                    width: "100%",
+                    fontSize: "0.875rem",
+                  }}
+                  variant="contained"
+                >
```

Comment:
If I'm correct the design styling of all the buttons throughout the app is the more or less the same . We can have a custom Styled Button in a separate styledComponents.ts file and use that styled button everywhere rather than rewriting the code for deisgn again and again. 
An example I can give is : 
`const StyledButton = styled(Button)((props) => ({
  borderWidth: pxToRem(3.5),
  borderColor: UIColors.primary.navyBlue,
  color: UIColors.primary.navyBlue,
  minHeight: pxToRem(44),
  fontWeight: 'bold',
  fontSize: pxToRem(18),
  '&:hover': {
    borderWidth: pxToRem(3.5),
    borderColor: UIColors.primary.navyBlue,
    color: UIColors.primary.navyBlue,
    cursor: 'pointer',
  },
  '&.Mui-disabled': {
    color: UIColors.primary.navyBlue,
    borderWidth: pxToRem(3.5),
    borderColor: UIColors.primary.navyBlue,
  },
}));`
If you see, the color values are coming from a constants file. The rem values are being calculated from a util function pxToRem that converts pixel to rem. Now that I have a styled button as I want in one place. I can use it in any component I like directly, rather than rewriting the code for it again.
This can be implemented for more components which are being repeated throughout your app.
Let's slowly try to implement this and improve the codebase quality.